activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -163,8 +163,18 @@ def polymorphic_options
           as_options(polymorphic).merge(options.slice(:null, :first, :after))
         end
 
-        def index_options
-          as_options(index)
+        def polymorphic_index_name(table_name)
+          "index_#{table_name}_on_#{name}"
+        end
+
+        def index_options(table_name)
+          index_options = as_options(index)
+
+          # legacy reference index names are used on versions 6.0 and earlier
+          return index_options if options[:uses_legacy_reference_index_name]
```

Comment:
Let's rename to `:_ uses_legacy_reference_index_name` to make explicit it is private.